### PR TITLE
[WIP] ci: Add arm64 runner to ci

### PR DESCRIPTION
**Pull request by jneo8**

### CODE REVIEW

**Review by aieri — APPROVED** [review]
lgtm aside from failing functional tests

**Comment by jneo8** [comment]
The CI failed because functional test try to deploy amd64 ubuntu charm, but we don't have this settings on the repo. Need to check details in zaza.

**Review by rgildein — CHANGES_REQUESTED** [review]
You are missing overlay for arch=arm64. The libjuju is using default value for architecture if it was not specified, as you can see [here](https://github.com/juju/python-libjuju/blob/b56963442a6deffe46ce1fb5cb41970344263649/juju/model.py#L1949C16-L1949C36).

File: Makefile [hunk 1]
```diff
@@ -44,6 +44,7 @@ submodules-update:
 	@git submodule update --init --recursive --remote --merge
 
 build: clean submodules-update
+	sudo apt -y install unzip
```
[the remark this answers is not in the document]
Comment (jneo8) [reply Makefile:47]: We can, but this is not the major task we want to handle here.

**Review by gabrielcocenza — COMMENTED** [review]
> The CI failed because functional test try to deploy amd64 ubuntu charm, but we don't have this settings on the repo. Need to check details in zaza why it's using amd64 as default.

For zaza you need to pass the env var MODEL_CONSTRAINTS and pass “arch=arm64”. Take a look at this example that I did during the [sprint](https://github.com/canonical/charmed-openstack-upgrader/pull/408/files)

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (chanchiwai-ray) [reply Makefile:47]: Why was this added? I don't see it was used anywhere in the PR. Even if we do use it somewhere, it's not ideal to put a `sudo xxx` in the `Makefile` as developer will use this in their local machine, and some could have a password-less `sudo`

**Review by chanchiwai-ray — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgildein) [reply Makefile:47]: There is unzip used later in Makefile. I'll suggest dropping that one also.

**Comment by rgildein** [comment]
> MODEL_CONSTRAINTS

The constraints could be set by [`model-constraints`](https://juju.is/docs/juju/juju-model-constraints) and zaza can configure it for us. Nice, I did not know about it :+1:

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgildein) [reply Makefile:46]: > We can, but this is not the major task we want to handle here.

So why you are adding it here, if it's not part of task.

**Pull request closed without merging**